### PR TITLE
SEO Draft

### PR DESCRIPTION
**Pull request by RealIndrit**
Yall will need to pull this in locally and test it, but I think it works?

### CODE REVIEW

**Comment by RealIndrit** [comment]
@jsweazy pleas take a look at this, caus I dont know what im doing, read docs, and think I did it right?

File: components/layout/PageLayout.tsx [hunk 1]
```diff
@@ -2,18 +2,38 @@ import clsx from 'clsx';
 
 import Footer from 'components/footer/Footer';
 import Header from 'components/header/Header';
+import {OpenGraph} from "../../model/opengraph";
```
Comment (jsweazy) [inline components/layout/PageLayout.tsx:5]:
```suggestion
import {OpenGraph} from "model/opengraph";
```

File: components/layout/PageLayout.tsx [hunk 2]
```diff
@@ -2,18 +2,38 @@ import clsx from 'clsx';
...
+          <meta property="og:image:alt" content={openGraph?.image?.url} />
+          <meta property="og:description" content={openGraph?.description} />
+          <meta property="og:site_name" content={openGraph?.site_name} />
+      </Head>
+        <div className="flex min-h-screen flex-col bg-knut-light-bg font-inter text-knut-light-text dark:bg-knut-dark-bg dark:text-knut-dark-text">
+          <Header />
+          <main className={clsx('container mx-auto flex-1 p-4', widthClassName)}>{children}</main>
+          <Footer />
+        </div>
+      </>
```
Comment (jsweazy) [inline components/layout/PageLayout.tsx:37]: indentation/spacing is off here

File: pages/content/index.tsx [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import PageLayout from 'components/layout/PageLayout';
 import { Playlist, PlaylistItem } from 'model/youtube';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
```
Comment (jsweazy) [inline pages/content/index.tsx:5]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { SEO } from "content/text";
import { baseUrl } from "content/links";
```

File: pages/contributors.tsx [hunk 4]
```diff
@@ -1,11 +1,20 @@
 import PageLayout from 'components/layout/PageLayout';
 import Important from 'components/ui/Important';
 import { contributors } from 'content/contributors';
+import {OpenGraph} from "../model/opengraph";
+import {seo} from "../content/text";
+import {baseUrl} from "../content/links";
```
Comment (jsweazy) [inline pages/contributors.tsx:6]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { SEO } from "content/text";
import { baseUrl } from "content/links";
```

File: pages/exercises/[slug].tsx [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import PageLayout from 'components/layout/PageLayout';
-import { placeHolderImage } from 'content/links';
+import {baseUrl, placeHolderImage} from 'content/links';
 import { IUIExercise } from 'model/ui';
 import { GetStaticPaths, GetStaticProps, InferGetStaticPropsType } from 'next';
 import { getExercises, getOneExercise } from 'utils/contentful';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
```
Comment (jsweazy) [inline pages/exercises/[slug].tsx:7]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
```

File: pages/exercises/[slug].tsx [hunk 6]
```diff
@@ -1,8 +1,10 @@
 import PageLayout from 'components/layout/PageLayout';
-import { placeHolderImage } from 'content/links';
+import {baseUrl, placeHolderImage} from 'content/links';
```
Comment (jsweazy) [inline pages/exercises/[slug].tsx:2]:
```suggestion
import { baseUrl, placeHolderImage } from 'content/links';
```

File: pages/exercises/[slug].tsx [hunk 7]
```diff
@@ -12,9 +14,17 @@ export default function ExercisePage({
...
+    title: name + " - "+  seo.title,
+    description: "An overview of exercises done during Camp Knut.",
+    type: "video.other",
+    url: baseUrl + "/exercises",
+    // cahnge  url to individual slug
+  }
 
   return (
-    <PageLayout>
+      <PageLayout openGraph={og}>
```
Comment (jsweazy) [inline pages/exercises/[slug].tsx:27]:
```suggestion
    <PageLayout openGraph={og}>
```

File: pages/exercises/index.tsx [hunk 8]
```diff
@@ -7,8 +7,10 @@ import { getExercises } from 'utils/contentful';
 import slugify from 'utils/slugify';
 import ScrollContainer from 'react-indiana-drag-scroll';
 import { IUIExercise } from 'model/ui';
-import { placeHolderImage } from 'content/links';
+import {baseUrl, placeHolderImage} from 'content/links';
```
Comment (jsweazy) [inline pages/exercises/index.tsx:10]:
```suggestion
import { baseUrl, placeHolderImage } from 'content/links';
```

File: pages/exercises/index.tsx [hunk 9]
```diff
@@ -7,8 +7,10 @@ import { getExercises } from 'utils/contentful';
...
 import { CgClose } from 'react-icons/cg';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
```
Comment (jsweazy) [inline pages/exercises/index.tsx:13]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
```

File: pages/index.tsx [hunk 10]
```diff
@@ -5,8 +5,10 @@ import Introduction from 'components/mainPage/Introduction';
 import Participants from 'components/mainPage/Participants';
 import CustomLink from 'components/ui/CustomLink';
 import Important from 'components/ui/Important';
-import { tips } from 'content/text';
+import {seo, tips} from 'content/text';
 import { Streamer } from 'model/twitch';
+import {OpenGraph, OpenGraphImage} from "../model/opengraph";
+import {baseUrl} from "../content/links";
```
Comment (jsweazy) [inline pages/index.tsx:11]:
```suggestion
import { OpenGraph, OpenGraphImage } from "model/opengraph";
import { baseUr } from "content/links";
```

File: pages/index.tsx [hunk 11]
```diff
@@ -5,8 +5,10 @@ import Introduction from 'components/mainPage/Introduction';
 import Participants from 'components/mainPage/Participants';
 import CustomLink from 'components/ui/CustomLink';
 import Important from 'components/ui/Important';
-import { tips } from 'content/text';
+import {seo, tips} from 'content/text';
```
Comment (jsweazy) [inline pages/index.tsx:8]:
```suggestion
import { seo, tips } from 'content/text';
```

File: pages/index.tsx [hunk 12]
```diff
@@ -45,8 +47,26 @@ interface HomeProps {
 }
 
 const Home = ({ twitch }: HomeProps) => {
+
```
Comment (jsweazy) [inline pages/index.tsx:50]:
```suggestion
```

File: pages/socials.tsx [hunk 13]
```diff
@@ -1,9 +1,17 @@
 import PageLayout from 'components/layout/PageLayout';
-import { socials } from 'content/links';
+import {baseUrl, socials} from 'content/links';
+import {OpenGraph} from "../model/opengraph";
+import {seo} from "../content/text";
```
Comment (jsweazy) [inline pages/socials.tsx:4]:
```suggestion
import { baseUrl, socials } from 'content/links';
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
```

File: pages/vote/index.tsx [hunk 14]
```diff
@@ -13,6 +13,9 @@ import {
   insertVoteForUser,
   getUsersVote,
 } from 'utils/supabaseClient';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
```
Comment (jsweazy) [inline pages/vote/index.tsx:18]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
import { baseUrl } from "content/links";
```

File: pages/workout/[slug].tsx [hunk 15]
```diff
@@ -5,10 +5,21 @@ import PageLayout from 'components/layout/PageLayout';
 import renderOptions from 'components/richTextStyling';
 import { IWorkoutPostFields } from 'model/contentful';
 import { getOneWorkoutPost, getWorkoutPosts } from 'utils/contentful';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
```
Comment (jsweazy) [inline pages/workout/[slug].tsx:10]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
import { baseUrl } from "content/links";
```

File: pages/workout/[slug].tsx [hunk 16]
```diff
@@ -5,10 +5,21 @@ import PageLayout from 'components/layout/PageLayout';
...
 
 export default function WorkoutPage({ post: { text, title } }: { post: IWorkoutPostFields }) {
+  let og:OpenGraph = {
+    site_name: seo.title,
+    title:  title + " - " + seo.title,
+    description: "Workout",
+    url: baseUrl + "/workout",
+    // cahnge  url to individual slug
```
Comment (jsweazy) [inline pages/workout/[slug].tsx:18]:
```suggestion
    // TODO:  change url to individual slug
```

File: pages/workout/index.tsx [hunk 17]
```diff
@@ -5,13 +5,24 @@ import Image from 'next/image';
 import PageLayout from 'components/layout/PageLayout';
 import { IWorkoutPost } from 'model/contentful';
 import { getWorkoutPosts } from 'utils/contentful';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
```
Comment (jsweazy) [inline pages/workout/index.tsx:10]:
```suggestion
import { OpenGraph } from "model/opengraph";
import { seo } from "content/text";
import { baseUrl } from "content/links";
```

File: pages/workout/index.tsx [hunk 18]
```diff
@@ -5,13 +5,24 @@ import Image from 'next/image';
...
 
 export default function WorkoutListPage({
   workouts,
 }: InferGetStaticPropsType<typeof getStaticProps>) {
+
```
Comment (jsweazy) [inline pages/workout/index.tsx:15]:
```suggestion
```

File: pages/workout/index.tsx [hunk 19]
```diff
@@ -5,13 +5,24 @@ import Image from 'next/image';
...
+  let og:OpenGraph = {
+    site_name: seo.title,
+    title: "Workout - " + seo.title,
+    description: "Workout plan for Camp Knut",
+    url: baseUrl + "/workout",
+  }
+
   return (
-    <PageLayout widthClassName="max-w-none">
+      <PageLayout widthClassName="max-w-none" openGraph={og}>
```
Comment (jsweazy) [inline pages/workout/index.tsx:25]:
```suggestion
    <PageLayout widthClassName="max-w-none" openGraph={og}>
```

File: utils/supabaseClient.ts [hunk 20]
```diff
@@ -1,8 +1,6 @@
 import { createClient, User } from "@supabase/supabase-js";
+import {baseUrl} from "../content/links";
```
Comment (jsweazy) [inline utils/supabaseClient.ts:2]:
```suggestion
import { baseUrl } from "content/links";
```

File: pages/content/index.tsx [hunk 21]
```diff
@@ -17,8 +20,16 @@ export async function getStaticProps() {
 }
 
 export default function ContentPage({ playlist }: { playlist: PlaylistItem[] }) {
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/content/index.tsx:23]:
```suggestion
  const og:OpenGraph = {
```

File: pages/contributors.tsx [hunk 22]
```diff
@@ -1,11 +1,20 @@
...
 
-const HallOfFame = () => {
+const Contributors = () => {
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/contributors.tsx:9]:
```suggestion
  const og:OpenGraph = {
```

File: pages/exercises/[slug].tsx [hunk 23]
```diff
@@ -12,9 +14,17 @@ export default function ExercisePage({
   const newUrl = new URL(youtubeLink);
   const videoId = newUrl.pathname.substring(1);
   const timestamp = newUrl.searchParams.get('t');
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/exercises/[slug].tsx:17]:
```suggestion
  const og:OpenGraph = {
```

File: pages/exercises/[slug].tsx [hunk 24]
```diff
@@ -12,9 +14,17 @@ export default function ExercisePage({
...
+    site_name: seo.title,
+    title: name + " - "+  seo.title,
+    description: "An overview of exercises done during Camp Knut.",
+    type: "video.other",
+    url: baseUrl + "/exercises",
+    // cahnge  url to individual slug
```
Comment (jsweazy) [inline pages/exercises/[slug].tsx:23]:
```suggestion
    // TODO: change url to individual slug
```

File: pages/exercises/index.tsx [hunk 25]
```diff
@@ -43,9 +45,15 @@ export default function ExerciseListPage({
     setSearchTags(searchTags.map((x) => ({ ...x, active: false })));
     setSearchTerm('');
   };
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/exercises/index.tsx:48]:
```suggestion
  const og:OpenGraph = {
```

File: pages/index.tsx [hunk 26]
```diff
@@ -45,8 +47,26 @@ interface HomeProps {
...
+  let og_Image:OpenGraphImage = {
```
Comment (jsweazy) [inline pages/index.tsx:51]:
```suggestion
  const og_Image:OpenGraphImage = {
```

File: pages/index.tsx [hunk 27]
```diff
@@ -45,8 +47,26 @@ interface HomeProps {
...
+    height: "300",
+    type: "image",
+    url: "",
+    width: "",
+    alt: "A surely not photoshoped/edited phot of Knut himself"
+  }
+
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/index.tsx:59]:
```suggestion
  const og:OpenGraph = {
```

File: pages/socials.tsx [hunk 28]
```diff
@@ -1,9 +1,17 @@
...
 
 export default function SocialsPage() {
+    let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/socials.tsx:7]:
```suggestion
    const og:OpenGraph = {
```

File: pages/vote/index.tsx [hunk 29]
```diff
@@ -79,6 +82,20 @@ export default function Vote() {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  let og_logged_in:OpenGraph = {
```
Comment (jsweazy) [inline pages/vote/index.tsx:85]:
```suggestion
  const og_logged_in:OpenGraph = {
```

File: pages/vote/index.tsx [hunk 30]
```diff
@@ -79,6 +82,20 @@ export default function Vote() {
...
+    site_name: seo.title,
+    title: "Vote - " + seo.title,
+    description: "Vote for the streamer you think has performed the best at Camp Knut",
+    url: baseUrl + "/vote",
+  }
+
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/vote/index.tsx:92]:
```suggestion
  const og:OpenGraph = {
```

File: pages/workout/[slug].tsx [hunk 31]
```diff
@@ -5,10 +5,21 @@ import PageLayout from 'components/layout/PageLayout';
 import renderOptions from 'components/richTextStyling';
 import { IWorkoutPostFields } from 'model/contentful';
 import { getOneWorkoutPost, getWorkoutPosts } from 'utils/contentful';
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
 
 export default function WorkoutPage({ post: { text, title } }: { post: IWorkoutPostFields }) {
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/workout/[slug].tsx:13]:
```suggestion
  const og:OpenGraph = {
```

File: pages/workout/index.tsx [hunk 32]
```diff
@@ -5,13 +5,24 @@ import Image from 'next/image';
...
+import {OpenGraph} from "../../model/opengraph";
+import {seo} from "../../content/text";
+import {baseUrl} from "../../content/links";
 
 export default function WorkoutListPage({
   workouts,
 }: InferGetStaticPropsType<typeof getStaticProps>) {
+
   const items = workouts.items as IWorkoutPost[];
+  let og:OpenGraph = {
```
Comment (jsweazy) [inline pages/workout/index.tsx:17]:
```suggestion
  const og:OpenGraph = {
```

**Review by jsweazy — COMMENTED** [review]
Pretty sure you don't have prettier working. There is a good amount of this prettier should have fixed for you on save.

**Pull request merged**